src/GraphQL/Types/Schema.cs

```diff
@@ -366,7 +366,14 @@ public void RegisterTypes(params Type[] types)
         /// <inheritdoc/>
         public void RegisterTypeMapping(Type clrType, Type graphType)
         {
-            (_clrToGraphTypeMappings ??= new()).Add((clrType ?? throw new ArgumentNullException(nameof(clrType)), graphType ?? throw new ArgumentNullException(nameof(graphType))));
+            (_clrToGraphTypeMappings ??= new()).Add((CheckType(clrType ?? throw new ArgumentNullException(nameof(clrType))), graphType ?? throw new ArgumentNullException(nameof(graphType))));
+
+            Type CheckType(Type clrType)
+            {
+                return typeof(IGraphType).IsAssignableFrom(clrType)
+                    ? throw new ArgumentException($"{clrType.FullName}' is already a GraphType (i.e. not CLR type like System.DateTime or System.String). You must specify CLR type instead of GraphType.", nameof(clrType))
```

Comment:
Fine with this check, but should be `ArgumentOutOfRangeException`.

```suggestion
                    ? throw new ArgumentOutOfRangeException(nameof(clrType), $"{clrType.FullName}' is already a GraphType (i.e. not CLR type like System.DateTime or System.String). You must specify CLR type instead of GraphType.", nameof(clrType))
```

Reply:
Should we add a check to be sure the graph type is indeed a graph type?

Reply:
I remoced second `nameof(clrType)`

> to be sure the graph type is indeed a graph type?

![изображение](https://user-images.githubusercontent.com/21261007/215894660-68307577-3b31-442b-b605-14003b6d7cc4.png)

?

Reply:
I mean a verification that `graphType` IS assignable from `IGraphType` within `RegisterTypeMapping`.

Reply:
> I removed second `nameof(clrType)`

:+1:

Reply:
I added second check. By the way yesterday I configured GPG signing so all my commits now are signed
![изображение](https://user-images.githubusercontent.com/21261007/215998072-10600b61-246b-411e-8ad7-74d8d7e81b65.png)
I thought it was time to do this, because I have to work not only with this project, but also others.